### ISSUE TITLE
Warning

### ISSUE THREAD

**Issue by maxozo** [broadinstitute/tensorqtl#43]
Hi, 
I am running tensorqtl cis mapping and getting the following warning. Would I need to be worried regarding this? 

_WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)_

Thanks,
Matiss

**Comment by francois-a** [comment]
Hi,
No, this isn't a problem if it only happens rarely.

**Comment by maxozo** [comment]
Thanks for the response. What if it happens quite often?

**Comment by francois-a** [comment]
The input VCF should only include common variants. The large number of monomorphic variants suggests that your input VCF was not filtered, and this is definitely not ideal (large numbers of rare variants will potentially lead to regression outlier artifacts).

**Comment by maxozo** [comment]
thanks francois-a. Really helpful.
What filters would you suggest to be applied to vcfs before running tensorQTL?

**Comment by francois-a** [comment]
In general sample sizes are not sufficient to map variants with MAF < 0.01. So applying a MAF filter and removing monomorphic variants should be part of initial QC. The MAF filter can be applied in-sample (with the `maf_threshold` option), but this won't remove monomorphic variants that are all, e.g., `0/1`.

**Comment by baishengjun** [comment]
I have filter the variants with MAF < 0.05, but I encounter too much the same warning above.
My sample size is only 38, does it cause this warning? and any solution to sovle this problem?

**Comment by Jesson-mark** [comment]
@francois-a 
Hi, 
I currently encounter the same warning but with different information when running tensorQTL with `cis.map_cis` command.

The logfile are:
```
/path/to/tensorqtl/core.py:315: RuntimeWarning: invalid value encountered in sqrt
  return 2*stats.t.cdf(-np.abs(np.sqrt(tstat2)), dof)
WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)
```

I already filtered my genotype file by MAF>0.01 so I didn't received warning about removing monomorphic variants. 
I'm running `cis.map_cis` to 15000 genes and this WARNING above occured 1286 times in my latest run. 

I'm wondering if this warning has some effect on the beta-approximated p-value of the 15000 genes and if it may reduce the power of discoverying significant eGenes? 

Any reply will be highly appreciated! 

JieWang

**Comment by francois-a** [comment]
Please try re-running this with v1.0.10, which includes an optimization that should get rid of (most of) these warnings.

**Comment by YimingHuang11** [comment]
I came across the same warning here for 866 out of 23620 phenotypes. For example,

` processing phenotype 29/22754WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)
/rds/general/user/xx/home/anaconda3/envs/env-qtl/lib/python3.8/site-packages/tensorqtl/core.py:315: RuntimeWarning: invalid value encountered in sqrt
  return 2*stats.t.cdf(-np.abs(np.sqrt(tstat2)), dof)
    processing phenotype 52/22754WARNING: scipy.optimize.newton failed to converge (running scipy.optimize.minimize)
/rds/general/user/xx/home/anaconda3/envs/env-qtl/lib/python3.8/site-packages/tensorqtl/core.py:315: RuntimeWarning: invalid value encountered in sqrt
  return 2*stats.t.cdf(-np.abs(np.sqrt(tstat2)), dof)
  `
I tried to pull out these phenotypes to see if any patterns can be found in common, but it was not obvious that they do not have necessarily low variance nor higher proportion of low expression. I wonder what could be the potential causes? Also, I removed these phenotypes (based on the number mentioned in 'processing phenotype 29/22754' as the row number in resulted cis-mapping table), but these errors reappeared for different phenotypes-- what is the order by which the phenotypes are processed?

The latest release version seems still v1.0.9 though?
